src/test/compile-fail/enum-and-module-in-same-scope.rs

```diff
@@ -13,7 +13,7 @@ mod Foo {
 }
 
 enum Foo {  //~ ERROR duplicate definition of type or module `Foo`
-    X //~ ERROR duplicate definition of value `X`
+    X
```

Comment:
I don't really know why this error went away, but it seems like it's probably fallout from some minor detail of how you are traversing? Or I guess we're not "merging" the enum/normal modules in some way that we used to do?